src/main/java/com/google/cloud/tools/project/ProjectIdValidator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.project;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Check if a string is a legal <a href='https://support.google.com/cloud/answer/6158840?hl=en'>Google 
+ * Cloud Platform Project ID</a>. Source: com.google.apphosting.base.AppId<p>
+ * 
+ * <pre>
+ * app-id ::= [(domain):](display-app-id)
+ * domain ::= r'(?!-)[a-z\d\-\.]{1,100}'
+ * display-app-id ::= r'(?!-)[a-z\d\-]{1,100}'
+ * </pre>
+ * 
+ * Note that in some older documentation this is referred to as an "application ID."
+ */
+public class ProjectIdValidator {
+
+  private static final Pattern PATTERN = Pattern.compile(
```

Comment:
The validation implemented in [cloud-tools-for-eclipse](https://github.com/GoogleCloudPlatform/cloud-tools-for-eclipse/blob/59f8eea489f68b849ddfda02b91a0a22ba8bd2f5/plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/databinding/ProjectIdValidator.java#L22) is rather different than this. Which one is correct?